### PR TITLE
 For #374: Implementing timeless user creation

### PR DESCRIPTION
**Pull request by paulodamaso**
For #374: Implementing timeless user creation:
- creates database if it does not exists
- creates user if it does not exists
- create Timeless company and user

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#457](https://github.com/timelesslounge/timelessis/issues/457) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#457](https://github.com/timelesslounge/timelessis/issues/457) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/17d3bebb-5a26-4994-bb99-ab0de67f472b); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: scripts/install/deploy/install_db.sh [hunk 1]
```diff
@@ -28,3 +28,39 @@ fi
...
+
+#checks if the database exists and create if not
+if [! psql -lqt | cut -d \| -f 1 | grep -qw "timelessdb"; ]; then
+    echo "Creating database: timelessdb"
+    sudo -u postgres psql -c "CREATE DATABASE timelessdb;"
+fi
+
+#checks if the user exists and create if not
+if [! psql -t -c '\du' | cut -d \| -f 1 | grep -qw "timeless_user"; ]; then
+    echo "Creating user: timeless_user"
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso can you add datetime to these echos? you did it already in the backup scripts, if I remember well, should be easy to take it from there

File: scripts/install/deploy/install_db.sh [hunk 2]
```diff
@@ -28,3 +28,39 @@ fi
 
 sudo cp timeless_pg.service /lib/systemd/system/
 sudo systemctl start timeless_pg.service
+
+#checks if the database exists and create if not
+if [! psql -lqt | cut -d \| -f 1 | grep -qw "timelessdb"; ]; then
+    echo "Creating database: timelessdb"
+    sudo -u postgres psql -c "CREATE DATABASE timelessdb;"
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso also add end logs, such as ``echo database created succesfully!``

File: scripts/install/deploy/install_db.sh [hunk 3]
```diff
@@ -28,3 +28,39 @@ fi
...
+#checks if the user exists and create if not
+if [! psql -t -c '\du' | cut -d \| -f 1 | grep -qw "timeless_user"; ]; then
+    echo "Creating user: timeless_user"
+    sudo -u postgres psql -c "CREATE USER timeless_user WITH
+        SUPERUSER
+        CREATEDB
+        CREATEROLE
+        INHERIT
+        LOGIN
+        ENCRYPTED PASSWORD 'timeless_pwd';"
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso is the ``;`` required in the SQL statemement?

File: scripts/install/deploy/install_db.sh [hunk 4]
```diff
@@ -28,3 +28,39 @@ fi
...
+fi
+
+company = $(sudo -u postgres -H -- psql -d timelessdb -c "INSERT INTO company
+(name, code, address) values ('Timeless', 'Tm', '')")
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso can you specify these SQL stements on one line each? I think it's better to have longer lines int his case, the statements are clearer.

File: scripts/install/deploy/install_db.sh [hunk 5]
```diff
@@ -28,3 +28,39 @@ fi
...
+role = $(sudo -u postgres -H -- psql -d timelessdb -c "INSERT INTO role
+(name, works_on_shifts, company_id) values ('Administrator', False, $company)")
+password = "pass from rultor"
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso ``pass from rultor``? Shouldn't there be an env. variable here or something?

File: scripts/install/deploy/install_db.sh [hunk 6]
```diff
@@ -28,3 +28,39 @@ fi
...
+sudo -u postgres -H -- psql -d timelessdb -c "INSERT INTO employee
+    (first_name, last_name, username, phone_number, birth_date,
+    registration_date, account_status, user_status, email, password, pin_code,
+    comment, company_id, role_id) values
+    ('First', 'Last', 'timeless', '988888', '', '', 'active', 'active',
+    'abc@xyz.com', $password, '10', 'Timeless user', $company, $role
```
Comment (amihaiemil) [inline scripts/install/deploy/install_db.sh]: @paulodamaso why mock data such as ``abc@xyz.com``?

Comment (paulodamaso) [reply scripts/install/deploy/install_db.sh]: @amihaiemil I don't have this data, I'll ask to @emilianodellacasa

Comment (paulodamaso) [reply scripts/install/deploy/install_db.sh]: @emilianodellacasa What values should we put here on this user account?

File: scripts/install/deploy/install_db.sh (hunk 5 above)
Comment (paulodamaso) [reply scripts/install/deploy/install_db.sh]: @amihaiemil We should get this value from rultor, I'll have to research about it

**Comment by paulodamaso** [comment]
@amihaiemil Done the corrections, and I have some comments, please take a look

File: scripts/install/deploy/install_db.sh (hunk 5 above)
Comment (amihaiemil) [reply scripts/install/deploy/install_db.sh]: @paulodamaso ok, then maybe you can leave a puzzle for the research, or?

**Comment by amihaiemil** [comment]
@paulodamaso ping, see my last question [here](https://github.com/timelesslounge/timelessis/pull/457#discussion_r262431374) :)

File: scripts/install/deploy/install_db.sh (hunk 6 above)
Comment (emilianodellacasa) [reply scripts/install/deploy/install_db.sh]: @paulodamaso Sorry, I lost your comments among the others: for this sensitive data, please use:


credentials.account.admin.first_name
credentials.account.admin.last_name
credentials.account.admin.username
credentials.account.admin.email
credentials.account.admin.password
credentials.account.admin.pincode

**Comment by paulodamaso** [comment]
@amihaiemil Updated the puzzle with more information about the task, please take a look

**Comment by amihaiemil** [comment]
@paulodamaso looks good, thanks!

**Comment by amihaiemil** [comment]
@rultor good to merge

File: scripts/install/deploy/install_db.sh [hunk 7]
```diff
@@ -1,8 +1,16 @@
 #!/bin/sh
 
 # @todo #299:30min Continue the deploy algorithm. We need to create an
```
Comment (emilianodellacasa) [inline scripts/install/deploy/install_db.sh]: @paulodamaso You should change the code of the puzzle, otherwise you are not removing it

**Comment by paulodamaso** [comment]
@emilianodellacasa I wonder why I keep forgetting this :) Corrected, please take a look

**Comment by emilianodellacasa** [comment]
@paulodamaso Maybe you didn't push the fix, I see the same code

**Comment by paulodamaso** [comment]
@emilianodellacasa Sorry! I've left home, and forgot to push the button

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#457](https://github.com/timelesslounge/timelessis/issues/457) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**